### PR TITLE
Experiments table refactoring

### PR DESCRIPTION
**Pull request by jrmartin**
Creating react component for Experiments table

### CODE REVIEW

File: src/main/webapp/js/components/dev/ExperimentsTable/.bower.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (slarson) [inline src/main/webapp/js/components/dev/ExperimentsTable/.bower.json]: @jrmartin why is there this file (.bower.json) as well as bower.json?  Are both necessary?

Comment (jrmartin) [inline src/main/webapp/js/components/dev/ExperimentsTable/.bower.json]: @slarson No, only bower.json, this was an extra hidden file that made it in

File: src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+		render: function() {
+			var id = "collapsable-"+this.props.experiment.getId();
+			var rowNumber = this.props.rowNumber;
+			var rowClasses = "nested-experiment-info"
+			//add different background to every even row
+			if(rowNumber%2==1){
+				rowClasses += " nthTr";
+			}
+			var style  = {
+				width:"215px"
```
Comment (slarson) [inline src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js]: Curious about this extra style information -- does it not work to have this in the less file?

File: src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+		render: function() {
+			var experiment = this.props.experiment;
+			//Create IDs for icons
+			var activeIconId = "activeIcon-" +this.props.experiment.getId();
+			var deleteIconId = "deleteIcon-" +this.props.experiment.getId();
+			var downloadResultsIconId = "downloadResultsIcon-" +this.props.experiment.getId();
+			var downloadModelsIconId = "downloadModelsIcon-" +this.props.experiment.getId();
+    
+			var style  = {
+					paddingRight:"10px",
```
Comment (slarson) [inline src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js]: Curious about this extra style information -- does it not work to have this in the less file?

File: src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+			var targetRowId = "."+rowID;
+			if( $(targetRowId).is(':visible')){
+				$(targetRowId).hide();
+			}else{
+				$(targetRowId).show();
+			}
+		},
+		render: function() {
+			var style  = {
+					width:"30%"
```
Comment (slarson) [inline src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js]: Curious about this extra style information -- does it not work to have this in the less file?

File: src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+					rows.push(<ExperimentExpandableRow experiment={experiment} rowNumber={this.state.counter} key={expandableRowId}/>);
+					this.state.counter++;
+				}
+			}.bind(this));
+
+					return (
+							<div className="col-lg-6 experimentsResults panel-body experimentsResultsOutput" id="experimentsOutput">
+								<table id="experimentsTable" className="table table-condensed experimentsTable" style={{borderCollapse:"collapse"}}>
+									<thead className="experimentsTableColumn">
+									<tr><th style={{width:"10%",textAlign:"center"}}>Status</th>
```
Comment (slarson) [inline src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js]: Curious about this extra style information -- does it not work to have this in the less file?

File: src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js (hunk 2 above)
Comment (slarson) [inline src/main/webapp/js/components/dev/ExperimentsTable/ExperimentsTable.js]: 👍  to that

**Comment by slarson** [comment]
@jrmartin Can you try replacing .travis.yml with the following and see if it enables both travis checks to work?

```
language: java
before_script:
- npm install -g bower
install:
#- git clone https://github.com/openworm/org.geppetto.model.git -b $TRAVIS_BRANCH
#- cd org.geppetto.model
#- mvn install
#- cd ..
#- git clone https://github.com/openworm/org.geppetto.core.git -b $TRAVIS_BRANCH
#- cd org.geppetto.core
#- mvn install
#- cd ..
#- git clone https://github.com/openworm/org.geppetto.simulation.git -b $TRAVIS_BRANCH
#- cd org.geppetto.simulation
#- mvn install
- cd ..
env:
  global:
    secure: EvqtUrZj3GsqzAkjZIVf2ZIze6ec72G8Dnkc0J3lKxwDTK6+KgIYgRWUWWO6tzL5Mn3Y+kkiHvyAkrHNPb2QZVBAdOdE3O308WRi64FF71nurd46zE43K/sSxcX/zCoBtxsz49oRcluPWl+nGDi/iKEysgcnQCvz23VWEYHAsOk=
```

**Comment by gidili** [comment]
@slarson @jrmartin reviewed the PR and ran the code

First of all great job on encapsulating the experiments table in various react components, looks very neat. 

As I started digging and running the code I found a few things worth discussing, here's my comments:

- [ ] React issue having to do with editable fields, we need to make that warning go away 
screenshot 2016-06-22 09 02 17
- [ ] The experiments tab is no longer resizable (it is on live)
- [ ] Column alignment could be improved / should not depend on length/lack of text (s-shot taken after creating new experiment). NOTE: this could be something that was already there, unrelated from the refactoring, not sure but it can be tested on OSB by creating a new experiment
screenshot 2016-06-22 09 08 08
- [ ] I tested with the persistence bundle and I cannot seem to be able to persist editable fields on the expandable rows anymore for an experiment in design status (simulator / timestep / length), editing works on the experiment name. In general we need to **make sure we do regression testing with the persistence bundle on**. This is kind of a show stopper - If this is a bug somewhere else and not related to the refactoring, let's create those cards but it needs to be investigated. Best case scenario this is my local problem... let me know ;)
- [ ] I noticed the CSS is in the less stuff instead of living with the component.The idea of these components is that they are self contained and they can be zipped up and uploaded so they can be included with bower / similar, so everything lives inside the component folder (see control panel or spotlight). If we split the css away from the component this is no longer possible. Also if the component is excluded from the components.json config file its css still gets deployed this way. In my opinion this is not ideal. Note: I know that historically we have some of the components css in the less stuff, so maybe this is not a big deal just a discussion point. Up to you guys, my suggestion is to keep the css with the component.
- [ ] Some code cleanup - always remember to do formatting with the IDE so that code formatting conventions for the language (in this case js) are followed. Plus I found a bunch of missing semicolons and extra semicolons (probably from copy-paste from the previous code that has been refactored). In generale this stuff shouldn't make it to the code review, but as I was going over it I did it and you can see results here (just formatted, adjusted spacing to be consistent and added/removed a bunch of semicolons) -->  https://github.com/openworm/org.geppetto.frontend/commit/f3a6781b6d05e4149c3614ce038af5c1ba5f6fa7

**Comment by jrmartin** [comment]
@gidili @slarson The persistence features work, but you have to press enter after you input the information on the field. It should have been the same behavior as before since the listener code is the same, but I might have missed an extra project save call that happens without pressing enter.

The warning on the console is a known bug https://github.com/facebook/draft-js/issues/81. 
We are running react 14.8, version 15 should take care of this warning showing up. 

I see the value  with the CSS lving with React component, maybe if we take that route of encasulpating all CSS within component we open a new card, more than one component makes references to classes in the less files. 

.

**Comment by slarson** [comment]
@tarelli this can be merged from now.  We addressed many of @gidili 's issues and created new cards for the ones we are leaving off for now.

**Comment by jrmartin** [comment]
@tarelli Development merged into this branch and tested, ready to be merged into development. Same has been done for core, simulation and persistence which have open pull request pending for experiments_table_refactoring

**Pull request merged**